packages/components/src/internal/components/search/SampleFinderSection.tsx

```diff
@@ -0,0 +1,183 @@
+import { SCHEMAS } from '../../schemas';
+import { InjectedQueryModels, withQueryModels } from '../../../public/QueryModel/withQueryModels';
+import { User } from '../base/models/User';
+import { SamplesEditableGridProps } from '../samples/SamplesEditableGrid';
+import { getFinderStartText } from './utils';
+
+const SAMPLE_FINDER_TITLE = "Find Samples";
+const SAMPLE_FINDER_CAPTION = "Find samples that meet all of these criteria";
+
+interface SampleFinderSamplesGridProps {
+    user: User;
+    getSampleAuditBehaviorType: () => AuditBehaviorTypes;
+    samplesEditableGridProps: Partial<SamplesEditableGridProps>;
+}
+
+interface Props extends SampleFinderSamplesGridProps {
+    parentEntityDataTypes: EntityDataType[];
+}
+
+interface SampleFinderHeaderProps {
+    parentEntityDataTypes: EntityDataType[],
+    onAddEntity: (entityType: EntityDataType) => void;
+}
+
+export const SampleFinderHeaderButtons: FC<SampleFinderHeaderProps> = memo(props => {
+    const { parentEntityDataTypes, onAddEntity } = props;
+
+    return (
+        <div>
+            Search by:
+            {parentEntityDataTypes.map((parentEntityType) => (
+                <button key={parentEntityType.nounSingular} className="btn btn-default margin-left" onClick={() => { onAddEntity(parentEntityType) }}>
+                    <i className="fa fa-plus-circle container--addition-icon" /> {capitalizeFirstChar(parentEntityType.nounAsParentSingular)} Properties
+                </button>
+            ))}
+        </div>
+    )
+});
+
+export const SampleFinderSection: FC<Props> = memo((props) => {
+    const { parentEntityDataTypes, ...gridProps } = props;
+
+    const [showFilterModal, setShowFilterModal] = useState<boolean>(false);
+    const [chosenEntityType, setChosenEntityType] = useState<EntityDataType>(undefined);
+    const [filterCards, setFilterCards] = useState<FilterCardProps[]>([]);
+
+    const onAddEntity = useCallback((entityType: EntityDataType) => {
+        setChosenEntityType(entityType);
+        setShowFilterModal(true);
+    }, []);
+
+    const onFilterEdit = useCallback((index: number) => {
+        console.log("onFilterEdit for index " + index + ": Not yet implemented.");
+        setShowFilterModal(true);
+    }, [filterCards]);
+
+    const onFilterDelete = useCallback((index: number) => {
+        const newFilterCards = [...filterCards];
+        newFilterCards.splice(index, 1);
+        setFilterCards(newFilterCards);
+    }, [filterCards]);
+
+    const onFilterClose = () => {
+        setShowFilterModal(false);
+    };
+
+    const onFind = useCallback((schemaQuery: SchemaQuery, filterArray: Filter.IFilter[])  => {
+        onFilterClose();
+        let newFilterCards = [...filterCards];
+        newFilterCards.push({
+            schemaQuery,
+            filterArray,
+            entityDataType: chosenEntityType,
+            index: filterCards.length,
```

Comment:
do we need to keep track of this `index`? can't it just be determined based on the index in the array? Maybe this is why the delete thing is removing more then one card as the index would change for the 3rd card if the 1st or 2nd was removed.